### DISCUSSION TITLE
Mobile Menu - Potential Issue

### DISCUSSION THREAD

**Discussion by southpaw3229** [templaza/astroid-framework#294] [General]
Love what you are doing with the framework.

One thing i've noticed though is the mobile menu is hard for users to navigate when you have sub headers.  Users click on the sub header to pop out the sub menu, but they have to click on the arrow.  I know it sounds like they should figure it out, but the mobile menu really isn't user friendly when you have sub menu headings.  Any thought on re-working the mobile menu?

Also i noticed on new iphone 14, if you have 7+ sub menus, it cuts off the 7th menu item and you can't scroll or get to it.

I would donate to this.

Also here is a screen video showing you with the sub headers, how it can be confusing to navigate.
https://watch.screencastify.com/v/jpUhoLEsdpDipboAI9Zy

Thanks

**Comment by Chacapamac** [comment]
Yep, this is something I fight for with the original Developer (Joomdev) 
**I should have point out this BIG problem long time ago.**
**This is a really bad behavior for the so important mobile menu.**

**I mark this as a bug as it render the mobile menu really difficult to use and really confusing for the user.**

**I expose the problem as images.** 

**OK BEHAVIOR**

![Screen Shot 2023-02-26 at 2 46 40 PM](https://user-images.githubusercontent.com/1907276/221433822-8f7b3034-c353-47a7-9686-b11373feb121.jpg)

**WRONG BEHAVIOR**

![Screen Shot 2023-02-26 at 2 58 39 PM](https://user-images.githubusercontent.com/1907276/221434949-d1cb87a7-f5ff-426e-9995-8cdfbfbbe082.jpg)

With Astroid now when clicking/taping the “**Alias**”, the menu close and go nowhere as see on the video cast. 
**Worst, Visitors probably see that as a bug to the website and abandon there research.**

![Screen Shot 2023-02-26 at 3 20 52 PM](https://user-images.githubusercontent.com/1907276/221435318-21ed391d-0701-4076-b333-d863bd6cb5b0.jpg)

**Look at my example of **“Alias menu”** opening the submenu (caret) correctly**
http://new-astroid.graf-test.ca/index.php

I just CSS the link from the caret all the way to the left and hide the gray separator between the “Alias” and the caret (important to make a distinction between a submenu with content on the left (not “Alias’ and submenu “Caret” link

**Comment by southpaw3229** [comment]
Thanks for the detailed workup.  Do you think there is any chance the new developer will spend time re-thinking the mobile navigation?  I have a lot of sites and don't want to hack/custom css each one unless I have to.

Thanks

**Comment by Chacapamac** [comment]
**@southpaw3229  as I consider this as a MAJOR bug, I’m persuaded that @sonvnn  will fix it.**

 I will personally insist and a little mad at myself to not have put that point to the front row as I knew it exist for a long time. 

The fact that I fix it by hacking the CSS, probably influx on the fact that I kind of forget that major point This is one of the first thing I fix on Astroid when the original developer Joomdev. Because the way I’m using Menu/submenu (Menu with submenu are never a link and ALWAYS an Alias)

**It’s simple, without my custom fix, I will never use Astroid in the first place as that behavior is unacceptable.**

**Comment by sonvnn** [comment]
Hello @southpaw3229 @Chacapamac 

Thanks for your feedback about UX. I improved mobile menu following this discussion on https://github.com/templaza/astroid-framework/commit/87ecdee6a21c127290150655ec6ea353edc1c1e6

You need download developer branch to test it.
To use this feature you have to change Menu Item Type to Menu Heading
![Screen Shot 2023-03-08 at 15 53 17](https://user-images.githubusercontent.com/3204103/223667549-61ec3721-52c3-477d-bde1-370a965c5be7.png)

Please help me test it and give me your comment about it.

Thanks & Best Regards,
Sonny

**Reply by Chacapamac** [reply]
@sonvnn 
**Is this implemented in Framewok 2.6.4 — Template one 2.6.3?**

**Comment by Chacapamac** [comment]
@sonvnn 
**Sorry it take me so much time to test it**

**This is my results, tell me what you think?**

![Separator](https://user-images.githubusercontent.com/1907276/228887908-e13b2b14-8ea5-41ba-87a4-03ba2158d4f0.jpg)

![Menu Heading](https://user-images.githubusercontent.com/1907276/228887956-1c594166-8943-4abf-89be-eff2a41c7cd0.jpg)

**Comment by Chacapamac** [comment]
I guess the line can be simply fix by css

**Reply by sonvnn** [reply]
Have a nice day my friend.
Yes, I will update this in the 2.6.6.

Best Regards,
Sonny

**Reply by Chacapamac** [reply]
Great, Good Day!
